### ISSUE TITLE
User scripts can no longer control logger outputs to files

### ISSUE THREAD

**Issue by bpbrown** [DedalusProject/dedalus#189]
In the past, it has been very handy to have user scripts specify logger output to files.  

With a code block like this:
```python
from dedalus.tools.config import config
config['logging']['filename'] = os.path.join(data_dir,'logs/dedalus_log')
config['logging']['file_level'] = 'DEBUG'

import dedalus.public as de
```
a user script will output debug level loggers to a a set of files in `data_dir/logs/dedalus_log_p#.log`, with `data_dir` a potentially script-specific location (e.g., a directory named with the parameters of a specific experiment).  Crucially, in this code block, `config` is imported from `dedalus.tools.config`, and relevant settings are made, before `dedalus.public` is imported.  `dedalus.public` in the past imported `from .tools import logging as _logging_setup`.

With commit 905fcff9a5c72fb18d1e28a367af8ec96150c7c5, this has changed, and the earlier implementation specifying logger output to files within user scripts now no longer works.

The key change appears to be the removal of 
```python
from .tools import logging as _logging_setup
```
from `public.py` and the inclusion of 
```python
# Import custom logging to setup rootlogger
from .tools import logging as _logging_setup
```
in `__init__.py`.

I have confirmed that user script logging control works in immediately preceding commit 030fa13cf2dddd2eb1c3c71ff7ce3b85125d3531, and I've tested the current master 76955c8df062e15164e73b38cf6d3a45a28a46e8, plus I have reverted these changes on the current master commit to confirm they are the source of the changed behaviour.

It's not clear to me how to solve this problem; adding the import from `.tools` block back into `public.py` still fails, and the only approach that has worked in my testing is to revert the changes in `__init.py__` (remove the import from `.tools`) and `public.py` (add the import from `.tools`) simultaneously.  But `__init__.py` needs this import to have a correctly performing logging for the threading warning.

**Comment by bpbrown** [comment]
MWE:

```python
import os

import logging
logger = logging.getLogger(__name__)

data_dir = 'test_case'

logger.info("this is a test script, writing to: {:}".format(data_dir))

from dedalus.tools.config import config
config['logging']['filename'] = os.path.join(data_dir,'logs/dedalus_log')
config['logging']['file_level'] = 'DEBUG'

if not os.path.exists('{:s}/'.format(data_dir)):
    os.mkdir('{:s}/'.format(data_dir))
logdir = os.path.join(data_dir,'logs')
if not os.path.exists(logdir):
    os.mkdir(logdir)

import dedalus.public as de
logger.debug("debug output test")
logger.info("info logger test")
logger.warning("warning test")
```

**Comment by kburns** [comment]
Hi Ben, I took a crack at it with 342f8ada7. I think the easiest approach is to make a function that adds the file handler, which we call at instantiation with the config from the config file, if set. And it can also be called after import to manually add a log from a script.  So now the syntax for a script could be like:

```
import numpy as np
import matplotlib.pyplot as plt
import dedalus.public as d3
import logging
logger = logging.getLogger(__name__)

import dedalus.tools.logging as dedalus_logging
dedalus_logging.add_file_handler('dedalus_log', 'DEBUG')
```

This is no longer sensitive to import orders, etc.  How does something like that sound?

**Comment by bpbrown** [comment]
Keaton, just tested out 342f8ada790d8690c96c6f16069a06feee27d1d1 and I think that's a really great solution!  As an added benefit, directory creation for the log is now handled automatically (that was a manual part of my user scripts before), so that removes an extra level of cruft and repeated code.

**Comment by kburns** [comment]
Fixed in #195, also backported to d2 in #193.
